### ISSUE TITLE
BED - Changes to the eBook Detail Page

### ISSUE THREAD

**Issue by janetgatsby** [patpluspun/fmp_au#77]
See attached image for changes to the eBook detail page template (individual book page). NOTE: this is for the title as appears in the EBOOK SECTION ONLY, no changes to the print version template (listing as the book appears in other sections of the website). 

@patpluspun please LOE this and ticket #76 also please let me know if FED is needed as well.

![ebook title page template changes](https://cloud.githubusercontent.com/assets/7650441/9294004/2e863d62-440d-11e5-898f-fc02b2c405dd.jpg)

**Comment by patpluspun** [comment]
I added a little bit of extra logic to make more sense:

If the stock is 0, and it has eBook as a catalog term, it uses the ebook template.  I didn't want to hide the purchase button in case there were any products left to be sold.

I added this overall logic to both display methods as well, so this ticket may rely on #76 for theming in case the missing components are problematic.  This also means that all eBooks with 0 stock will adapt this template.  This also means that any items listed in eBooks that have purchasable stock will retain the old template.

I will ask Emily about this while you're on vacation, and send her a link to look at the existing changes for approval.

**Comment by janetgatsby** [comment]
@patpluspun yes, please talk to Emily. I'm not sure what you did was necessary. The book can exist as an ebook and printed book (with inventory) at the same time. What she wants to do is only show the ebooks options when you are in the ebooks section of the site http://www.fivemile.com.au/catalog/ebooks-0 (hide purchasing the physical book). But when you are in any other section of the site, you can order the physical book and/or the ebook.

**Comment by janetgatsby** [comment]
Patrick, hope this will help clarify:

In the workflow where a user visits the eBook landing page and then clicks on a title to view the detail page - Only show the eBook purchasing information for the title.

In the workflow where a user visits a book detail page via any other method than described above, display both Print and Ebook purchasing options.

**Comment by patpluspun** [comment]
This will work on the catalog pages, and I've already set up a new view to handle ebook page to show the correct template.

However, there is no real distinction between an ebook and a physical book as far as the catalog is concerned.  It doesn't know where the user came from to determine the correct template on the book detail page.  To do so would require adding an extra option for "Ebook" and removing the term from the catalog.  Then the template would apply only to books with an Ebook check, but there would still be the issue of determining which template to display if a book has an ebook and physical presence.  The only way I can see that happening is to literally build another catalog for ebooks only, that will ensure that catalog uses the ebook book detail template.

It won't be a super huge change, but would require a lengthy update to fix all the old products.

**Comment by janetgatsby** [comment]
Yes, that is what I was trying to get at. It would essentially be a change to how ebooks are handled by building another catalog. 

What do you think is the best approach (and the easiest for content entry) for adding a new book title that has both a print catalog entry and an ebook catalog entry? 

How would we handle the ISBN# on the ebook catalog? It might not need one, but would be good to use to identify relationship to the print version. Maybe adding an "e" to the string? I'm suggesting this because I'm assuming using the ISBN on two catalog entries would be a conflict. If I'm incorrect then disregard this part of the question.

**Comment by patpluspun** [comment]
The easiest approach would be to simply build an entirely new catalog from Ebooks.  I don't think they would need an ISBN unless there were a print version as well, but the current system requires a unique ISBN of some sort.

The most accurate way to do it would be to create a new content type, but to prevent from having to create two products I'd have to generate ebooks from a product programmatically, and also generate ones for the existing products with ebook category.  This would be a lot of work on my end, but it's doable.  By the next release, anyway, not tonight :P

**Comment by janetgatsby** [comment]
There will always be a print version for a corresponding ebook. They don't do any exclusively ebook titles.
I agree with your proposed approach and I'll put this in the next milestone.

Is there any way I can help with the generation of existing products once you put the solution in place?

**Comment by patpluspun** [comment]
Not really, it will all be automated, so it will just take time.

**Comment by janetgatsby** [comment]
@patpluspun this is closed but I don't see the changes to the book detail page on test.
Example: http://test.fivemile.com.au/catalog/adult/men-menu

**Comment by janetgatsby** [comment]
@patpluspun I do not see this change on production. Do you need to run the generation of existing products?

http://www.fivemile.com.au/catalog/new-releases/rugby-league-rivalries
